explore_data/cluster_by_accessory_gene.py

```diff
@@ -0,0 +1,609 @@
+pao1_acc = list(set(pao1_ref_genes) - set(my_core_pao1_genes))
+print(f"Number of PAO1-specific genes: {len(pao1_acc)}")
+
+## TO DO: Get PA14-specific genes
+pa14_acc = list(set(pa14_ref_genes) - set(my_core_pa14_genes))
+print(f"Number of PA14-specific genes: {len(pa14_acc)}")
+
+# ## Create df for plotting
+
+# +
+# Merge strain to expression matrix
+pao1_expression_label = pao1_expression.merge(
+    sample_to_strain_table, left_index=True, right_index=True
+)
+pa14_expression_label = pa14_expression.merge(
+    sample_to_strain_table, left_index=True, right_index=True
+)
+
+pao1_expression_label.head()
+
+# +
+# Create core dataframe with columns:
+# core gene ids | median core expression | strain label
+
+# PAO1
+pao1_core_expression_label = pao1_expression_label[my_core_pao1_genes + ["Strain type"]]
+pao1_core_expression = pao1_expression_label[my_core_pao1_genes]
+pao1_core_expression_label["median core expression"] = pao1_core_expression.median(
+    axis=1
+)
+
+# PA14
+pa14_core_expression_label = pa14_expression_label[my_core_pa14_genes + ["Strain type"]]
+pa14_core_expression = pa14_expression_label[my_core_pa14_genes]
+pa14_core_expression_label["median core expression"] = pa14_core_expression.median(
+    axis=1
+)
+
+pao1_core_expression_label.head()
+# -
+
+pa14_core_expression_label.head()
+
+# +
+# Merge PAO1 and PA14 core dataframes
+pao1_pa14_core_expression_label = pao1_core_expression_label.merge(
+    pa14_core_expression_label,
+    left_index=True,
+    right_index=True,
+    suffixes=["_pao1", "_pa14"],
+)
+
+pao1_pa14_core_expression_label.head()
+
+# +
+# Create accessory df
+# accessory gene ids | median accessory expression | strain label
+
+# PAO1
+pao1_acc_expression_label = pao1_expression_label[pao1_acc + ["Strain type"]]
+pao1_acc_expression = pao1_expression_label[pao1_acc]
+pao1_acc_expression_label["median acc expression"] = pao1_acc_expression.median(axis=1)
+
+# PA14
+pa14_acc_expression_label = pa14_expression_label[pa14_acc + ["Strain type"]]
+pa14_acc_expression = pa14_expression_label[pa14_acc]
+pa14_acc_expression_label["median acc expression"] = pa14_acc_expression.median(axis=1)
+
+pao1_acc_expression_label.head()
+# -
+
+pa14_acc_expression_label.head()
+
+# +
+# Merge PAO1 and PA14 accessory dataframes
+pao1_pa14_acc_expression_label = pao1_acc_expression_label.merge(
+    pa14_acc_expression_label,
+    left_index=True,
+    right_index=True,
+    suffixes=["_pao1", "_pa14"],
+)
+
+pao1_pa14_acc_expression_label.head()
+# -
+
+# ## Plot: core genome
+
+# Drop label columns
+pao1_expression_numeric = pao1_expression_label.drop(["Strain type"], axis=1)
+pa14_expression_numeric = pa14_expression_label.drop(["Strain type"], axis=1)
+
+# +
+# Embed expression data into low dimensional space
+pca = PCA(n_components=2)
+# model_pao1 = umap.UMAP(random_state=123).fit(pao1_expression_numeric)
+model_pao1 = pca.fit(pao1_expression_numeric)
+pao1_encoded = model_pao1.transform(pao1_expression_numeric)
```

Comment:
in the PCA fit are the values normalized in anyway? Looking at the PCA plot, I'm wondering if it's being driven by a few genes that have very high expression.

Reply:
These are great points. I think I'll try a couple of things here: 1) normalize and re-cluster and 2) remove other strain types and re-cluster